x-pack/plugins/integration_assistant/server/graphs/log_type_detection/csv.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { LogFormatDetectionState } from '../../types';
+import type { LogDetectionNodeParams } from './types';
+import { createJSONInput } from '../../util';
+import { createCSVProcessor, createDropProcessor } from '../../util/processors';
+import { CSVParseError, UnparseableCSVFormatError } from '../../lib/errors/unparseable_csv_error';
+
+// We will only create the processor for the first MAX_CSV_COLUMNS columns.
+const MAX_CSV_COLUMNS = 100;
```

Comment:
Should this change once we have chunk in place? Or we move from `10` to a bigger number in the UI?

Reply:
That's the maximum number of columns, not rows, so it's not related to chunking. Realistically people don't use CSV when there's more than like 20–30 columns, so we don't expect this to ever be relevant. If anyone thinks anyone will create a CSV with 101 column, sure, let's set a higher number.